### PR TITLE
DB-8752 Delete obsolete statistics entries when truncating tables

### PR DESCRIPTION
**Pull request by ascend1**

### CODE REVIEW

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by dgomezferro** [comment]
@carolp-503 do we want to remove the already leaked entries with an upgrade script here or as part of the splice-ck effort ?

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1362/

**Comment by ascend1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1370/

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/TruncateTableConstantOperation.java [hunk 1]
```diff
@@ -225,9 +226,12 @@ private void truncateTable(TableDescriptor td, Activation activation) throws Sta
             }
         }
 
+        // Remove statistics
+        dd.deleteColumnStatistics(oldHeapConglom, tc);
+        dd.deletePartitionStatistics(oldHeapConglom, tc);
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/TruncateTableConstantOperation.java:231]: DataDictionaryImpl.deletePartitionStatistics() will delete both table and column stats, so `dd.deleteColumnStatistics(oldHeapConglom, tc);` is redundant.

Comment (ascend1) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/TruncateTableConstantOperation.java:230]: Done.

**Comment by jpanko1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1407/

**Comment by jpanko1** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**